File: LeetCode/1-Two-Sum.ts
function twoSum(nums: number[], target: number): number[] {
    let map : {[key: number]: number} = {}
    for(let i = 0; i < nums.length; i++)
    {
        
        if(nums[i] in map){
        return [i, map[nums[i]]];
        }
        map[target - nums[i]] = i;
    }
    return [-1, -1];
};

File: 1-Two-Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> ht = new Dictionary<int, int>();
        
        for (int i = 0; i < nums.Length; i++) {
            int tmp = nums[i];
            
            if (ht.ContainsKey(target - tmp)) {
                return new int[] {ht[target - tmp], i};
            }
            
            ht[tmp] = i;
        }
        
        return new int[0];
    }
}